Shift3/boilerplate-client-react
# Refactor to move `HolyGrailLayout` component to `common/components` folder

## Changes
1. Moved type definitions to the same file as the `HolyGrailLayout` component.
2. Moved `HolyGrailLayout` from `features/auth/components` to `common/components`.
3. Updated import paths to use new path for `HolyGrailLayout`.

## Purpose
The `HolyGrailLayout` is an app-wide shared component and not part of a specific feature. Thus, it should be in the `common/components folder`.

## Testing
1. Pull in the changes to your local copy of this branch.
2. Start the app with `yarn start`.
3. Log in to the app.
4. Ensure that the app still renders the navbar, content, and footer using the holy grail layout.
5. Ensure that toggling the navbar still toggles between rendering the navbar at the top and the side.

### Closes #478
